apps/voting/app/src/App.js

```diff
@@ -260,7 +243,65 @@ class App extends React.Component {
   }
 }
 
-export default observe(
-  observable => observable.map(state => ({ ...state })),
-  {}
-)(App)
+export default observe(observable => {
+  return observable.map(state => {
+    const settingsLoaded = hasLoadedVoteSettings(state)
+    if (!settingsLoaded) {
+      return {
+        ...state,
+        settingsLoaded,
+      }
+    }
+
+    const { pctBase, supportRequiredPct, tokenDecimals, voteTime } = state
+
+    const pctBaseNum = parseInt(pctBase, 10)
+    const supportRequiredPctNum = parseInt(supportRequiredPct, 10)
+    const tokenDecimalsNum = parseInt(tokenDecimals, 10)
+    const tokenDecimalsBaseNum = Math.pow(10, tokenDecimalsNum)
+
+    return {
+      ...state,
+
+      settingsLoaded,
+      pctBase: new BN(pctBase),
+      supportRequiredPct: new BN(supportRequiredPct),
+      tokenDecimals: new BN(tokenDecimals),
+
+      numData: {
+        pctBase: pctBaseNum,
+        supportRequiredPct: supportRequiredPctNum,
+        tokenDecimals: tokenDecimalsNum,
+      },
+
+      // Transform the vote data for the frontend
+      votes:
+        state && state.votes
+          ? state.votes.map(vote => {
+              const { data } = vote
+              return {
+                ...vote,
+                data: {
+                  ...data,
+                  endDate: new Date(data.startDate + voteTime),
+                  minAcceptQuorum: new BN(data.minAcceptQuorum),
+                  nay: new BN(data.nay),
+                  totalVoters: new BN(data.totalVoters),
+                  yea: new BN(data.yea),
+                  supportRequiredPct: new BN(supportRequiredPct),
+                },
+                numData: {
```

Comment:
Not the nicest way of doing it... but saves us from having to convert back and forth since the majority of the calculations don't need the accuracy of BN.js.